uriget/uriget.go

```diff
@@ -233,3 +239,38 @@ func (o *options) getGit(ctx context.Context, u *url.URL) ([]byte, error) {
 	o.logger.Printf("Read %d bytes from %s", len(buff), filepath.Join(td, subPath))
 	return buff, nil
 }
+
+func (o *options) getOci(ctx context.Context, u *url.URL) ([]byte, error) {
+	parts := strings.Split(u.Host+u.Path, "/")
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("invalid OCI URL format")
+	}
+	registry := parts[0]
+	repo := strings.Join(parts[1:len(parts)-1], "/")
+	tag := "latest"
+	lastPart := parts[len(parts)-1]
+	if strings.Contains(lastPart, ":") {
+		split := strings.Split(lastPart, ":")
+		repo = strings.Join(parts[1:len(parts)-1], "/") + "/" + split[0]
+		tag = split[1]
+	}
+	store, err := oci.New(o.tempDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OCI layout store: %w", err)
+	}
+	repoUrl := fmt.Sprintf("%s/%s", registry, repo)
+	remoteRepo, err := remote.NewRepository(repoUrl)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to remote repository: %w", err)
+	}
+	if strings.HasPrefix(repoUrl, "localhost:") || strings.HasPrefix(repoUrl, "127.0.0.1:") {
```

Comment:
Do we really need this? Genuine question, I am not sure about it, it sounds weird to me that this is not handled by oras.

Reply:
I added it cause I was pulling a local image from my docker remote repo,  Basically, the terminal kept complaining about the HTTP request so I had to filter it, to avoid the errors and to pull the image, Maybe it was no use , so I'll remove it

Reply:
I wasn't able to find anything like that in all the examples from oras library ([here](https://pkg.go.dev/oras.land/oras-go/v2#example-Copy-LocalToLocal) some examples) but I'm not an expert here, as said, it's only a gut feeling that this might be not required :).

Reply:
And yes you are right :)